src/input/input_python.rs

```diff
@@ -352,6 +354,56 @@ impl<'a> Input<'a> for PyAny {
         }
     }
 
+    fn strict_decimal(&'a self, decimal_type: &'a PyType) -> ValResult<&'a PyAny> {
+        let py = decimal_type.py();
+
+        // Fast path for existing decimal objects
+        if self.is_exact_instance(decimal_type) {
+            return Ok(self);
+        }
+
+        // Try subclasses of decimals, they will be upcast to Decimal
```

Comment:
Any reason to not allow instances of subclasses of decimal through without an upcast?

Reply:
In general this is what we've decided as the treatment of subclasses for V2 (e.g. int, float subclasses also upcast). In a future release maybe we will change this.